integration/spark/src/main/scala/org/apache/spark/sql/execution/command/index/DropIndexCommand.scala

```diff
@@ -210,12 +210,6 @@ private[sql] case class DropIndexCommand(
           logError("Table metadata unlocking is unsuccessful, index table may be in stale state")
         }
       }
-      // in case if the the physical folders still exists for the index table
-      // but the carbon and hive info for the index table is removed,
-      // DROP INDEX IF EXISTS should clean up those physical directories
-      if (ifExistsSet && carbonTable.isEmpty) {
```

Comment:
i think we should keep this change, from the comment its clear that there will be such scenarios and we have observed many times

Reply:
create table t1...
create table t2...
drop index t2 on t1

this drop index sql will remove table folder of t2 by mistake.

Reply:
are you saying that, even though the t2 is not an index, and i call drop index on t2, it will consider as a index and drop t2 table folder? If so, we should handle there and keep this code here. This was handled specially for SI i think in negative scenarios

Reply:
I agree with @akashrn5 : If code is buggy, we need to fix it. If we remove it. It might bring back the original issues that caused this code change.